src/plugins/intel_gpu/src/graph/reshape.cpp

```diff
@@ -207,6 +207,9 @@ void reshape_inst::update_output_memory() {
         return;
 
     build_deps();  // reshape need deps
+    if (node->get_program().get_config().get_property(ov::intel_gpu::allow_new_shape_infer) &&
+        input_memory_ptr() == nullptr)
+        return;
```

Comment:
No clear why the reshape's input memory is nullptr too. Could you clarify what was the case?

Reply:
The nodes in this model are connected as follows:
Loop (dynamic) -> Gather (dynamic) -> Reshape1 (static, opt out) -> Reshape2 (static, opt out) -> Concat (static)

The ticket issue is produced in Reshape2:
As Reshape1 is opt out and its input memory (Loop) is dynamic, its output is not assigned.
And Reshape2 checks input memory when it assigns output memory to input memory.
And as Reshape1 output memory is not assigned, Reshape2 generates an exception when it checks it.

Reply:
So the target crash happens at build time?

Reply:
Yes

Reply:
It is the line 213 in this file.
```cpp
213    OPENVINO_ASSERT(input_memory_ptr() != nullptr, "[GPU] Failed to reuse input in ", id(), " primitive: input memory was not allocated");
214    _outputs = {_network.get_engine().reinterpret_buffer(input_memory(), _impl_params->get_output_layout())};
}
```